crates/matrix-sdk-ui/src/room_list_service/state.rs

```diff
@@ -57,10 +61,18 @@ impl State {
 
             SettingUp | Recovering => {
                 set_all_rooms_to_growing_sync_mode(sliding_sync).await?;
-                Running
+                Running { last_time: Instant::now() }
             }
 
-            Running => Running,
+            Running { last_time } => {
+                // We haven't sync for a while so we should go back to recovering
+                if last_time.elapsed().as_secs() > 1800 {
```

Comment:
Ideally, we would like to make this (`1800`) configurable because it can depend on the homeserver or how the user wants to use this API. Thoughts?

Reply:
If this is configurable, then we don't need `mock_instant`, nor the `testing` feature (we want to remove them, and this patch is adding them). If this is configurable then, we can simply set it to 1 sec, and have a `sleep` of 2 secs in the test. It can sound like a bit hacky but it's probably easier to maintain than having a new dependency and a new feature.

Reply:
I am not sure it will be used much but it doesn't cost much to make it a parameter, and if it helps removing some maintenance burden in the tests it's probably worth it.

Yeah what you propose for the test is a bit hacky but it works :tm: , and it feels fine at the integration level.

On it.

Reply:
Some docs are still missing, I'll clean up later on in the week, probably from Berlin ;)